### PR TITLE
Consolidating rule IDs for invoking and referencing insecure deserialization, and a little bit of renumbering

### PR DESCRIPTION
**Pull request by dotpaul**
- Old CA2300 and CA2301 -> New CA2300
- Old CA2302 -> New CA2301
- Old CA2303 -> New CA2302
- Old CA2310 and CA2311 -> New CA2305

### CODE REVIEW

**Comment by dotpaul** [comment]
FYI @timhermann @matt-desai @LingxiaChen @dsie63

File: src/Microsoft.NetCore.Analyzers/Core/Security/DoNotUseInsecureDeserializerBinaryFormatterWithoutBinder.cs [hunk 1]
```diff
@@ -17,14 +17,14 @@ public class DoNotUseInsecureDeserializerBinaryFormatterWithoutBinder : DoNotUse
         // TODO paulming: Help link URLs.
         internal static readonly DiagnosticDescriptor RealBinderDefinitelyNotSetDescriptor =
             SecurityHelpers.CreateDiagnosticDescriptor(
-                "CA2302",
+                "CA2301",
```
Comment (mavasani) [inline src/Microsoft.NetCore.Analyzers/Core/Security/DoNotUseInsecureDeserializerBinaryFormatterWithoutBinder.cs]: Are these all new rules? As in will it break any existing suppressions for these IDs by existing binary FxCop customers?

Comment (dotpaul) [reply src/Microsoft.NetCore.Analyzers/Core/Security/DoNotUseInsecureDeserializerBinaryFormatterWithoutBinder.cs]: Yep, all new rules.  That's why I wanted to get this change in now, before we ship anything.

**Comment by dotpaul** [comment]
Thank you @mavasani !

**Pull request merged**